astropy/coordinates/builtin_frames/utils.py

```diff
@@ -236,25 +279,61 @@ def atciqz(rc, dc, astrom):
     same approach used by the ERFA functions above, except that they use a threshold of
     9 arcseconds.
 
+    In addition, ERFA's atciqz assumes a distant source, so there is no difference between
+    the object-Sun vector and the observer-Sun vector. This can lead to errors of up to a
+    few arcseconds in the worst case (e.g a Venus transit).
+
     Parameters
     ----------
-    rc : float or `~numpy.ndarray`
-        right ascension, radians
-    dc : float or `~numpy.ndarray`
-        declination, radians
+    astrometric : `~astropy.coordinates.CartesianRepresentation`
+        Astrometric ICRS position of object from observer
     astrom : eraASTROM array
         ERFA astrometry context, as produced by, e.g. ``eraApci13`` or ``eraApcs13``
 
     Returns
     --------
     ri : float or `~numpy.ndarray`
```

Comment:
Since we are changing the input to be a representation, should we also change the output? Maybe a `UnitSphericalRepresentation`?

Reply:
I think this would add a conversion step and make the results harder to deal with. Currently the ra, dec output is pretty easy to handle afterwards - i.e adding distance if necessary